### PR TITLE
Ubuntu 24.04

### PR DESCRIPTION
**Pull request by maxime-desroches**
todo: 
~opencl runtime~
~devcontainer~
~need to build malibre-native-qt libs on aarch64 on 24.04~
~make metadrive work  -> kind of done~
~test for building on 20.04 in CI~

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -330,8 +331,17 @@ qt_flags = [
   "-DQT_MESSAGELOGCONTEXT",
 ]
 qt_env['CXXFLAGS'] += qt_flags
-qt_env['LIBPATH'] += ['#selfdrive/ui', f"#third_party/maplibre-native-qt/{arch}/lib"]
-qt_env['RPATH'] += [Dir(f"#third_party/maplibre-native-qt/{arch}/lib").srcnode().abspath]
+
+maplibre_path = f"#third_party/maplibre-native-qt/{arch}/lib"
+if arch == "x86_64":
+  if UBUNTU_FOCAL:
```
Comment (adeebshihadeh) [inline SConstruct:337]: this is a bit janky. another option is to disable building the map stuff: https://github.com/commaai/openpilot/blob/149238845c1ed98dc5b86446c5c6d49e53b5e340/selfdrive/ui/SConscript#L12

**Comment by adeebshihadeh** [comment]
I upgraded my workstation at home to 24.04 and didn't notice any issues!

**Comment by adeebshihadeh** [comment]
Once this is merged, let's also make sure to update the containers in `xx/chffr/`.

File: .github/workflows/tools_tests.yaml [hunk 2]
```diff
@@ -39,12 +39,13 @@ jobs:
                         source selfdrive/test/setup_vsound.sh && \
                         CI=1 pytest tools/sim/tests/test_metadrive_bridge.py"
 
-  test_python311:
-    name: test python3.11 support
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+  test_compatibility:
+    name: test ubuntu/python support
```
Comment (adeebshihadeh) [inline .github/workflows/tools_tests.yaml:43]:
```suggestion
    name: test 20.04 + 3.11
```

File: pyproject.toml [hunk 3]
```diff
@@ -176,7 +176,7 @@ flaky = "*"
 inputs = "*"
 lru-dict = "*"
 matplotlib = "*"
-metadrive-simulator = { git = "https://github.com/commaai/metadrive.git", branch = "python3.12", markers = "platform_machine != 'aarch64'" } # no linux/aarch64 wheels for certain dependencies
+metadrive-simulator = { git = "https://github.com/commaai/metadrive.git", branch = "anisotropic_off", markers = "platform_machine != 'aarch64'" } # no linux/aarch64 wheels for certain dependencies
```
Comment (adeebshihadeh) [inline pyproject.toml:179]: we should make a PR to expose this. ideally we don't have to maintain a fork long term

**Review by adeebshihadeh — APPROVED** [review]

File: .github/workflows/tools_tests.yaml [hunk 4]
```diff
@@ -57,6 +58,10 @@ jobs:
       run: pip install poetry==1.7.0
     - name: Installing python dependencies
       run: poetry install --no-cache --no-root
+    - name: git LFS
+      run: git lfs pull
+    - name: Building openpilot
+      run: poetry run scons -u -j$(nproc)
```
Comment (adeebshihadeh) [inline .github/workflows/tools_tests.yaml:64]: this needs a cache. takes too long from scratch every time

File: .github/workflows/tools_tests.yaml [hunk 5]
```diff
@@ -58,8 +58,15 @@ jobs:
       run: pip install poetry==1.7.0
     - name: Installing python dependencies
       run: poetry install --no-cache --no-root
+    - uses: ./.github/workflows/setup-with-retry
+      with:
+        build_docker: 'false'
     - name: git LFS
       run: git lfs pull
+    - name: Get scons cache
```
Comment (adeebshihadeh) [inline .github/workflows/tools_tests.yaml:66]: does the scons cache match between 20.04 and 24.04? i would think it needs its own cache

Comment (maxime-desroches) [reply .github/workflows/tools_tests.yaml:66]: there was some cache hits: https://github.com/commaai/openpilot/actions/runs/9556512327/job/26341924114?pr=32624#step:11:34

will check how much it match

Comment (adeebshihadeh) [reply .github/workflows/tools_tests.yaml:66]: the Cython stuff for sure won't match

**Pull request merged**